models/_sector/nft/trades/ethereum/platforms/superrare_ethereum_base_trades.sql

```diff
@@ -0,0 +1,227 @@
+        and topic1 = lower('0xea6d16c6bfcad11577aef5cc6728231c9f069ac78393828f8ca96847405902a9')
+        {% if is_incremental() %}
+        and block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        and block_time >= '{{ project_start_date }}'
+        {% endif %}
+
+    union all
+
+    -- Superrare Marketplace (not decoded)
+    select block_time
+            , block_number
+            , concat('0x',substring(topic2 from 27 for 40)) as contract_address
+            , bytea2numeric_v3(substring(data from 67 for 64)) as nft_token_id
+            , concat('0x',substring(topic4 from 27 for 40)) as seller
+            , concat('0x',substring(topic3 from 27 for 40)) as buyer
+            , bytea2numeric_v3(substring(data from 3 for 64)) as price_raw
+            , tx_hash
+            , '{{ var("ETH_ERC20_ADDRESS") }}'
+            , index
+    from {{ source('ethereum','logs') }}
+    where contract_address =  lower('0x65b49f7aee40347f5a90b714be4ef086f3fe5e2c')
+        and topic1 in (lower('0x2a9d06eec42acd217a17785dbec90b8b4f01a93ecd8c127edd36bfccf239f8b6')
+                        , lower('0x5764dbcef91eb6f946584f4ea671217c686fa7e858ce4f9f42d08422b86556a9')
+                      )
+        {% if is_incremental() %}
+        and block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        and block_time >= '{{ project_start_date }}'
+        {% endif %}
+)
+
+SELECT
+    'ethereum' as blockchain,
+    'superrare' as project,
+    'v1' as version,
+    cast(date_trunc('day', a.block_time) AS date) AS block_date,
+    a.block_time,
+    a.block_number,
+    a.nft_token_id,
+    CAST(1 AS DECIMAL(38,0)) as nft_amount,
+    'Buy' as trade_category,
+    a.seller,
+    a.buyer,
+    CAST(a.price_raw AS DECIMAL(38,0)) as price_raw,
+    a.currency_contract,
+    a.contract_address as nft_contract_address,
+    cast(NULL as varchar(1)) as project_contract_address,
+    a.evt_tx_hash as tx_hash,
+    case
+        when a.seller = coalesce(minter.to, minter_superrare.to)
+        then 'primary'
+        else 'secondary'
+    end as trade_type,
+    case
+        when a.seller = coalesce(minter.to, minter_superrare.to)
+        then cast(ROUND((0.03+0.15) * (a.price_raw)) as decimal(38)) -- superrare takes fixed 3% fee + 15% commission on primary sales
+        else cast(ROUND(0.03 * (a.price_raw)) as decimal(38))    -- fixed 3% fee
+    end as platform_fee_amount_raw,
+    case
+        when a.seller = coalesce(minter.to, minter_superrare.to)
+        then cast(0 as decimal(38))
+        else cast(ROUND(0.10 * (a.price_raw)) as decimal(38))  -- fixed 10% royalty fee on secondary sales
+    end as royalty_fee_amount_raw,
+    cast(NULL as varchar(1)) as royalty_fee_address,
+    cast(NULL as varchar(1)) as platform_fee_address,
+    sub_tx_trade_id
+from all_superrare_sales a
+left join {{ source('erc721_ethereum','evt_transfer') }} minter on minter.contract_address = a.contract_address
```

Comment:
do these have to be left join, rather than inner?

Reply:
These have to be left joins as superare has non-erc721 tokens.